pkg/controller/user/user_controller.go

```diff
@@ -173,36 +177,39 @@ func (c *userController) reconcile(key string) error {
 	} else {
 		// The object is being deleted
 		if sliceutil.HasString(user.ObjectMeta.Finalizers, finalizer) {
-
-			klog.V(4).Infof("delete user %s", key)
 			// we do not need to delete the user from ldapServer when ldapClient is nil
 			if c.ldapClient != nil {
-				if err = c.ldapClient.Delete(key); err != nil && err != ldapclient.ErrUserNotExists {
+				if err = c.waitForDeleteFromLDAP(key); err != nil {
 					klog.Error(err)
+					c.recorder.Event(user, corev1.EventTypeWarning, controller.FailedSynced, fmt.Sprintf("Failed to sync: %s", err))
 					return err
 				}
 			}
 
 			if err = c.deleteRoleBindings(user); err != nil {
 				klog.Error(err)
+				c.recorder.Event(user, corev1.EventTypeWarning, controller.FailedSynced, fmt.Sprintf("Failed to sync: %s", err))
```

Comment:
The message "Failed to sync: %s" is used in many places, consider to define it as constrant.

Reply:
fixed